src/persist-client/src/write.rs

```diff
@@ -48,7 +48,7 @@ use crate::ShardId;
 /// tokio::time::timeout(timeout, write.fetch_recent_upper()).await
 /// # };
 /// ```
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
oh please no. this will make things very difficult for us if we need to bring back writer registrationpush the resulting future into

Reply:
I should have mentioned that this PR is quite experimental, sorry about that! :see_no_evil: Some of the things are not meant to stay like this, if the PR goes in at all.